src/lightning-address.ts

```diff
@@ -0,0 +1,155 @@
+import fetch from 'cross-fetch';
+import { isUrl, isValidAmount, parseLnUrlPayResponse } from './utils/lnurl';
+import Invoice from './invoice';
+import { InvoiceArgs, ZapArgs } from './types';
+import { generateZapEvent } from './utils/nostr';
+
+const LN_ADDRESS_REGEX =
+/^((?:[^<>()\[\]\\.,;:\s@"]+(?:\.[^<>()\[\]\\.,;:\s@"]+)*)|(?:".+"))@((?:\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(?:(?:[a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+
+const DEFAULT_PROXY = "https://lnaddressproxy.getalby.com/lightning-address-details";
+
+export default class LightningAddress {
+  address: string;
+  options: { proxy: string };
+  username: string | undefined;
+  domain: string | undefined;
+  pubkey: string | undefined;
+  keysendData: unknown;
+  lnurlpData: Record<string, any>;
+
+  constructor(address: string, options: { proxy: string }) {
+    this.address = address;
+    this.options = { proxy: DEFAULT_PROXY };
+    this.options = Object.assign(this.options, options);
+    this.parse();
+    this.lnurlpData = {};
+  }
+
+  parse() {
+    const result = LN_ADDRESS_REGEX.exec(this.address.toLowerCase());
+    if (result) {
+      this.username = result[1];
+      this.domain = result[2];
+    }
+  }
+
+  async fetch() {
+    if (this.options.proxy) {
+      return this.fetchWithProxy();
+    } else {
+      return this.fetchWithoutProxy();
+    }
+  }
+
+  async fetchWithProxy() {
+    const result = await fetch(`${this.options.proxy}?${new URLSearchParams({ln: this.address}).toString()}`);
+    const json = await result.json();
+    this.lnurlpData = json.lnurlp;
+    this.keysendData = json.keysend;
+  }
+
+  async fetchWithoutProxy() {
+    try {
+      const lnurlResult = await fetch(this.lnurlpUrl());
+      this.lnurlpData = await lnurlResult.json();
+    } catch(e) {
+    }
+    try {
+      const keysendResult = await fetch(this.keysendUrl());
+      this.keysendData = await keysendResult.json();
+    } catch(e) {
+    }
+  }
+
+  lnurlpUrl() {
+    return `https://${this.domain}/.well-known/lnurlp/${this.username}`;
+  }
+
+  keysendUrl() {
+    return `https://${this.domain}/.well-known/keysend/${this.username}`;
+  }
+
+  async generateInvoice(url: URL): Promise<Invoice> {
+    const data = await fetch(url)
+    const json = await data.json()
+    const paymentRequest = json && json.pr && json.pr.toString()
+    if (!paymentRequest) throw new Error('Invalid pay service invoice')
+
+    const invoiceArgs: InvoiceArgs = {pr: paymentRequest};
+    if (json && json.verify) invoiceArgs.verify = json.verify.toString();
+
+    return new Invoice(invoiceArgs);
+  }
+
+  async requestInvoice(
```

Comment:
should this method also be used by the zap() call? 

this validates the amounts for example (which is currently not done in the zap() function)